src/irr/asset/CPLYMeshFileLoader.cpp

```diff
@@ -577,42 +577,55 @@ bool CPLYMeshFileLoader::genVertBuffersForMBuffer(asset::ICPUMeshBuffer* _mbuf,
 				check = _attribs[i].size();
 		}
 	}
+	
 	auto putAttr = [&_attribs](asset::ICPUMeshBuffer* _buf, size_t _attr)
 	{
 		size_t i = 0u;
 		for (const core::vectorSIMDf& v : _attribs[_attr])
 			_buf->setAttribute(v, _attr, i++);
 	};
 
-	size_t sizes[4];
+	uint32_t sizes[4];
 	sizes[E_POS] = !_attribs[E_POS].empty() * 3 * sizeof(float);
 	sizes[E_COL] = !_attribs[E_COL].empty() * 4 * sizeof(float);
 	sizes[E_UV] = !_attribs[E_UV].empty() * 2 * sizeof(float);
 	sizes[E_NORM] = !_attribs[E_NORM].empty() * 3 * sizeof(float);
 
-	size_t offsets[4]{ 0u };
+	uint32_t offsets[4]{ 0u };
 	for (size_t i = 1u; i < 4u; ++i)
 		offsets[i] = offsets[i - 1] + sizes[i - 1];
 
-	const size_t stride = std::accumulate(sizes, sizes + 4, static_cast<size_t>(0));
+	const uint32_t vertexSize = std::accumulate(sizes, sizes + 4, static_cast<size_t>(0));
 
 	{
-		const static std::array<std::pair<uint16_t, E_FORMAT>, 4> perIndexDataThatChanges{std::make_pair(E_POS, asset::EF_R32G32B32_SFLOAT), std::make_pair(E_COL, asset::EF_R32G32B32A32_SFLOAT), std::make_pair(E_UV, asset::EF_R32G32_SFLOAT), std::make_pair(E_NORM, asset::EF_R32G32B32_SFLOAT)};
-		_mbuf->setVertexBufferBinding(std::move(bufferBinding), 0ull);
-		_mbuf->setVertexBufferBindingParams(0ull, stride);
-
-		for (const auto& attributeIndexExtra : perIndexDataThatChanges)
-			[&](auto attribIndex, auto formatToSend, auto offsetIndex, auto bindingIndex)
+		const static std::array<std::pair<uint16_t, E_FORMAT>, 4> perIndexDataThatChanges{ std::make_pair(E_POS, asset::EF_R32G32B32_SFLOAT), std::make_pair(E_COL, asset::EF_R32G32B32A32_SFLOAT), std::make_pair(E_UV, asset::EF_R32G32_SFLOAT), std::make_pair(E_NORM, asset::EF_R32G32B32_SFLOAT) };
+		auto mbPipeline = _mbuf->getPipeline();
+		auto inputParams = mbPipeline->getVertexInputParams();
+		inputParams.enabledAttribFlags = 0b1111u;
```

Comment:
build them correctly out of `E_POS`, `E_COL` , etc. using `<<` and `|`

Reply:
okay